var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -138,6 +155,13 @@ class Charliecloud(AutotoolsPackage):
     # See https://github.com/spack/spack/pull/16049.
     conflicts("platform=darwin", msg="This package does not build on macOS")
 
+    def url_for_version(self, version):
+        if version >= Version("0.39"):
+            url_fmt = "https://gitlab.com/charliecloud/main/-/archive/v{0}/main-v{0}.tar.gz"
```

Comment:
This is the automated tarball from GitLab, which doesn’t contain a pre-built `configure`, etc. The old GitHub URLs are the release tarballs we make. I haven’t yet figured out how to add release tarball to GitLab.

Reply:
You can define `force_autoreconf = True` or `@property def force_autoreconf(self):` to have it depend on `self.spec.satisfies(@0.30:)`